### PR TITLE
show default values in help for integers and non-empty strings

### PR DESCRIPTION
**Pull request by harmti**
For example:
```
> python -m aiven.client service logs --help
...
   -S {desc,asc}, --sort-order {desc,asc}
                        Sort direction for log fetching (default: asc)
   -n LIMIT, --limit LIMIT
                        Get up to N rows of logs (default: 100)
...
```

### CODE REVIEW

File: aiven/client/argx.py [hunk 1]
```diff
@@ -33,6 +33,22 @@ def default(self, o):  # pylint:disable=E0202
         return jsonlib.JSONEncoder.default(self, o)
 
 
+class CustomFormatter(argparse.ArgumentDefaultsHelpFormatter, argparse.RawDescriptionHelpFormatter):
+    """Help formatter to display the default value only for integers and non-empty strings"""
+    def _get_help_string(self, action):
```
Comment (mwfrojdman) [inline aiven/client/argx.py]: Why is `ArgumentDefaultsHelpFormatter` inherited? It only defines the `_get_help_string` method, which this class overrides. Both superclasses inherit from `HelpFormatter`.

File: aiven/client/argx.py [hunk 2]
```diff
@@ -33,6 +33,22 @@ def default(self, o):  # pylint:disable=E0202
...
+        help_text = action.help
+        if '%(default)' not in action.help:
+            if action.default is not argparse.SUPPRESS:
+                defaulting_nargs = [argparse.OPTIONAL, argparse.ZERO_OR_MORE]
```
Comment (mwfrojdman) [inline aiven/client/argx.py]: What's `defaulting`? Could just be inline in the `if` statement below?

File: aiven/client/argx.py [hunk 3]
```diff
@@ -33,6 +33,22 @@ def default(self, o):  # pylint:disable=E0202
...
+                if action.option_strings or action.nargs in defaulting_nargs:
+                    if (
```
Comment (mwfrojdman) [inline aiven/client/argx.py]: All four nested if statements could be combined into one. If that is too unwieldy, consider using `if all([<first if condition>, <second if condition>, <third>, <fourth>]):`

File: aiven/client/argx.py (hunk 2 above)
Comment (mwfrojdman) [reply aiven/client/argx.py]: Never mind, misread that and understood that it's a default for `nargs`, not if a default help text should be used for certain `nargs` values.

File: aiven/client/argx.py (hunk 1 above)
Comment (harmti) [reply aiven/client/argx.py]: Dropped ArgumentDefaultsHelpFormatter

File: aiven/client/argx.py (hunk 3 above)
Comment (harmti) [reply aiven/client/argx.py]: Simplified, but still 3 separate if clauses left

**Pull request merged**